src/frontend/src/optimizer/mod.rs

```diff
@@ -392,12 +393,6 @@ impl BatchPlanRoot {
         // Convert to distributed plan
         plan = plan.to_distributed_with_required(&self.required_order, &self.required_dist)?;
 
-        // Add Project if the any position of `self.out_fields` is set to zero.
-        if self.out_fields.count_ones(..) != self.out_fields.len() {
-            plan =
-                BatchProject::new(generic::Project::with_out_fields(plan, &self.out_fields)).into();
-        }
-
```

Comment:
Why do we need to move this?

Reply:
Because the order column might be be included in the output, but it might still be used in the later in `BatchExchange` if `require_additional_exchange_on_root_in_distributed_mode`. This is also the handling order for local mode. It might be a bug previously, but just rarely be hit.